### ISSUE TITLE
0.8 beta failure

### ISSUE THREAD

**Issue by ghost** [vsespb/mt-aws-glacier#14]
Just trying out 0.8 beta. I gather that the journal data is now being stored as metadata on glacier. 

I'm getting this error when running:
./mtglacier.pl sync --config=/root/glacier.cfg --from-dir /mnt/media/video/Clips/ --to-vault=Clips --journal=/root/glacier-journal.log --concurrency=3 

"encode_base64url" is not exported by the MIME::Base64 module
 "decode_base64url" is not exported by the MIME::Base64 module
Can't continue after import errors at MetaData.pm line 28
BEGIN failed--compilation aborted at MetaData.pm line 28.
Compilation failed in require at GlacierRequest.pm line 31.
BEGIN failed--compilation aborted at GlacierRequest.pm line 31.
Compilation failed in require at ChildWorker.pm line 24.
BEGIN failed--compilation aborted at ChildWorker.pm line 24.
Compilation failed in require at ./mtglacier.pl line 35.
BEGIN failed--compilation aborted at ./mtglacier.pl line 35.

Should I not be specifying the journal on the command line as it's now being stored on glacier?  How should I get my existing journal onto glacier, or do I have to reupload everything again?

**Comment by vsespb** [comment]
i am currently offline till 2nd Jan. try updating mime::base64. journal is
still needed. you can use old journal with new version. i ll get back to
you later.
 30.12.2012 22:03 ÐÏÌØÚÏ×ÁÔÅÌØ "telscuba" notifications@github.com
ÎÁÐÉÓÁÌ:

**Comment by vsespb** [comment]
So..

Function encode_base64url was added to MIME::Base64 library in y2010.
I removed this dependency now (version 0.81beta).

You can now use 0.81 with your existing Perl distribution without updating anything.

> Should I not be specifying the journal on the command line as it's now being stored on glacier?

You still should specify Journal.

It's needed because Amazon Glacier inventory (which contains all archive ids and metadata) is not available in realtime.
To retrieve inventory you need 4hours (if last change to vault
was made 24+hours ago, otherwise you need 4+24hours and few requests).

So to determine which files are in sync and which are not, you need to have Journal.

BTW any existing Amazon Glacier client implementation use some kind of metadata cache/log/journal etc

Since version 0.8 mt-aws-glacier stores metadata (file names and modification times) in both journal and Amazon Glacier metadata.
So if you loose your journal file (created after v0.8) it won't be a disaster anymore, since it can be retrieved from Amazon Glacier (however withing 4+24hours)

(note: restoring journal from Amazon Glacier inventory is not implemented yet, but will be done very soon).

> How should I get my existing journal onto glacier, or do I have to reupload everything again?

Existing journal file will work with current code without problems. Old records will remain in old format. New records will be in a new format.
If you loose that your journal you'll loose metadata for older (0.7x) records but will be able to restore metadata for new records (0.8x)

So, basically you have two options:

1) continue using older journal and backup the journal
2) re-upload data and there will be no need to backup journal (however it's still better to backup it, just in case).

Note: I am planning to implement option to restore some metadata from inventory and some from old journal backups, so if you choose (1) it will be possible to backup old journal just once.

Let me know if this helps.

**Comment by ghost** [comment]
It helps immensely.  I understand the problems and the solutions, and I
thank you for this software!
